lib/server.js

```diff
@@ -0,0 +1,65 @@
+const app = require('express')();
+const redis = require('redis');
+const bodyParser = require('body-parser').json();
+const shortid = require('shortid');
+const url = require('url');
+
+const API_HOST = process.env.WHISPER_HOST || 'localhost';
+const API_PORT = process.env.WHISPER_PORT || 5000;
+
+// create a new redis client and connect to our local redis instance
+const REDIS_PORT = process.env.REDIS_PORT || 6379;
+const REDIS_HOST = process.env.REDIS_HOST || 'localhost';
+const client = redis.createClient(REDIS_PORT, REDIS_HOST);
+
+client.on('error', err => {
+  console.log("Error " + err);
+});
+client.on('error', error => console.error(`redis error: ${error}`));
+client.on('end', () => console.log('Redis: connection closed'));
+client.on('warning', msg => console.log(`Redis warning: ${msg}`));
+
+app.set('host', API_HOST);
+app.set('port', API_PORT);
+
+app.get('/whisper/:id', (req, res) => getWhisper(req, res, client));
+app.post('/whisper', bodyParser, (req, res) => postWhisper(req, res, client));
+
+app.listen(app.get('port'), () => {
+  console.log('Server listening on port: ', app.get('port'));
+});
+
+function getWhisper(req, res, client) {
+  const id = req.params.id;
+
+  client.get(id, (err, result) => {
+    if(err) {console.error(err);}
+
+    res.set('content-type', 'application/json');
+
+    res.status(200).send(result.toString());
+  });
+}
+
+function postWhisper(req, res, client) {
+  const whisper = req.body;
+  console.log('Incoming post:', whisper);
+
+  const id = shortid.generate();
+
+  console.log('Saving to:', id);
+
+  // TODO: validate input
+  client.set(id, JSON.stringify(whisper), 'EX', whisper.expiration);
+
+  res.status(201).send(whisperUrl(id));
+}
+
+function whisperUrl(id) {
```

Comment:
in deployment, the server needs to return https://whisper.demo.veres.one/whisper/foo without port numbers and using https protocol.  Need the magic to make that happen.  Here's how it's done in bedrock, maybe that convention is worth following: https://github.com/digitalbazaar/bedrock-server/blob/master/lib/config.js#L12

Reply:
I should add, that this is the case when the server is operating on HTTP port N

Reply:
And to be even more clear, we do *not* need the whisper server itself to operate using SSL.

Reply:
Ahh, good point, will fix.

Reply:
> And to be even more clear, we do not need the whisper server itself to operate using SSL.

Note, some imminent version of Chrome is going to start marking all HTTP sites as non-secure with big red X's or something. So, going forward, every site "needs" to be HTTPS.

Reply:
To clarify, this is because the server is deployed behind a reverse proxy that handles the SSL bits.